### DISCUSSION TITLE
exec cluster join "failed to ensure vela-system namespace..."

### DISCUSSION THREAD

**Discussion by hanyan007** [kubevela/kubevela#5122] [Q&A] [answered]
**Describe the problem**
Can't add cluster 
exex like:  vela  cluster join ~/.kube/staging-config --name  cluster-staging

reply: Error: failed to ensure vela-system namespace installed in cluster cluster-staging: failed to check if namespace vela-system exists: the server is currently unable to handle the request (get namespaces vela-system)

**ENV：**
vela version
CLI Version: v1.6.2
Core Version:
GitRevision: git-b1cc06b
GolangVersion: go1.19.2

CentOS Linux release 7.8.2003 (Core)

Cluster: aliyun ACK/ASK

**vela and the ACK nodes in the same VPC network.**

**TEST Join ACK Cluster**
![image](https://user-images.githubusercontent.com/14324188/203914241-f0c27a16-8e76-4116-b877-a1b5e339968a.png)

**TEST Join ASK Cluster, Use Public IP**
![image](https://user-images.githubusercontent.com/14324188/203914866-24cea05e-1e59-4a29-bf13-971234d21441.png)

And There are no errors in the apiserver log.

i read the issue of [4607](https://github.com/kubevela/kubevela/issues/4607)，but no help.

please help me，thanks very much.

**Comment by wonderflow** [comment, marked as answer]
1. Does the control plane cluster has the accessbility to the managed cluster?
2. How do you install kubevela control plane? Are you using velad ?

**Reply by hanyan007** [reply]
1,Does the  control plane cluster, means the local cluster？ may not be able to access the  managed  cluster.

we have some k8s clusters, like:
 test env ACK cluster (installed kubevela and some addons)
 staging env ACK cluster
 prod env ACK cluster
and dev env  ASK cluster (Public IP)
and some OPS ECSs ，which one installed vela command
They are all in one VPC environment,  The clusters cannot access each other. but the OPS ECS can，We use kubectl to maintain these clusters with inner network.

2, not the velad, installed like this:
curl -fsSl https://kubevela.net/script/install.sh |bash

**Reply by wonderflow** [reply]
Yes, the local cluster must have access to these managed clusters.

**Reply by hanyan007** [reply]
I solved this problem.

because we  used the “云效-APPSTACK“ last week ，and did not clean up completely.
after reinstall kubeveal, it‘s work.
thank you.
![image](https://user-images.githubusercontent.com/14324188/203953474-0907f60b-095f-4885-92c3-2ebb56edf94b.png)

**Reply by wonderflow** [reply]
great
